console/client/src/features/requests/RequestsModal.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react'
+import { useClient } from '../../hooks/use-client.ts'
+import { ConsoleService } from '../../protos/xyz/block/ftl/v1/console/console_connect.ts'
+import { Call } from '../../protos/xyz/block/ftl/v1/console/console_pb.ts'
+import { formatDuration, formatTimestamp } from '../../utils/date.utils.ts'
+import {  useSearchParams, useNavigate , useLocation } from 'react-router-dom'
+import { Dialog, Transition } from '@headlessui/react'
+import { ChevronRightIcon } from '@heroicons/react/20/solid'
+import { RequestGraph } from './RequestGraph.tsx'
+
+export function RequestModal() {
+  const navigate = useNavigate()
+  const location = useLocation()
+  const [ searchParams ] = useSearchParams()
+  const client = useClient(ConsoleService)
+  const [ calls, setCalls ] = React.useState<Call[]>([])
+  const key = searchParams.get('requests')?? undefined
+  const moduleName = searchParams.get('details')
```

Comment:
Honestly I consider most of this stuff code we'll discard, so not big. 

I do have a generic version I made but realized but headless is also a generic library so my efforts were pointless and also made things more complicated. 

I'll create some mixins object and function for applying css and other properties. We'd also benefit from 2 declarative components for background. Other than that I'd hold off on a pure Modal component. It would lead to prop creep as you continue to try and handle differing requirements. From my experience making design systems it's better to just have nice css & js mixins and very few declarative helpers components like what I described above.